website/docs/getting-started/setup/instance-configuration/http-proxy.md

```diff
@@ -0,0 +1,34 @@
+---
+description: This page provides steps to configure HTTP/HTTPS proxy on your self-hosted instance.
+---
+
+# Configure Proxy
+
+Appsmith offers support for running behind a forward proxy, which is beneficial when deploying Appsmith on a server behind a corporate firewall, and allows internet access through a proxy server. This page provides instructions for configuring HTTP/HTTPS proxy on your self-hosted Appsmith instance.
+
+## Prerequisites
+
+- A self-hosted Appsmith instance. See the [installation guides](/getting-started/setup/installation-guides) to set up your Appsmith instance.
+
+## Configure HTTP/HTTPS proxy
+
+Follow these steps to configure HTTP/HTTPS proxy for your self-hosted instance:
+
+1. Go to your Appsmith instance configuration file, such as the `docker.env` file for Docker or the `values.yaml` file for Kubernetes.
+2. Use any one of the below pair of environment variables to set proxy:
+
+ a. `HTTP_PROXY` and/or `HTTPS_PROXY` 
+
+ b. `http_proxy` and/or `https_proxy`
+
+3. Add the chosen environment variables as shown below:
+ ```bash
+ HTTP_PROXY=http://1.2.3.4:8080
+ HTTPS_PROXY=http://1.2.3.4:8080
+ ```
+4. Save the changes and restart the Appsmith instance.
+5. To use an HTTPS proxy, you also need to add a trusted custom Certificate Authority (CA) to Appsmith. See [Custom Root CA Certificate](/getting-started/setup/instance-configuration/custom-domain/custom-ca-root-certificate) to set up a certificate authority.
```

Comment:
This looks more like a prerequisite to me

Reply:
Sure, we can have it before, HTTPS proxies can be used without needing a custom CA, I think, but it's unusual to use it that way. And will most likely throw errors. I don't think we should do it as a prerequisite.

Reply:
@sharat87 - Does this mean that:
* Either this is more like a recommended setting, or this can serve as a troubleshooting guide to fix an error that a user might see.

Reply:
No, I don't think this should be a recommendation, since such a recommendation should come in from their company policy that required the proxy in the first place. I wouldn't presume to make a recommendation here.

I also don't think it should go in troubleshooting guide, cause, I don't want them to go see the error, then come back, look around in the troubleshooting, and then figure it out. This is common enough that we should frame it something like:

> When using an `HTTPS_PROXY`, you'll usually also want to add a custom CA to be trusted. If this is the case for your proxy, please check out this other document.

Or something similar.